File: content/LLD/Reference-Code/4. Observer Pattern/main.py
from weather_station import WeatherStation
from tv import TVDisplay
from mobile import MobileDisplay

ws = WeatherStation()
tv = TVDisplay()

ws.add_observer(tv)
ws.update_temprature(30)

mobile = MobileDisplay()
ws.add_observer(mobile)
ws.update_temprature(35)

ws.remove_observer(tv)
ws.update_temprature(40)

File: content/LLD/Reference-Code/4. Observer Pattern/Main.java
public class Main {
    public static void main(String[] args) {
        WeatherStation ws = new WeatherStation();
        TvDisplay tv = new TvDisplay();

        ws.addObserver(tv);
        ws.updateTemperature(30);

        MobileDisplay mobile = new MobileDisplay();
        ws.addObserver(mobile);
        ws.updateTemperature(35);

        ws.removeObserver(tv);
        ws.updateTemperature(40);
    }
}
